code/cmdline/cmdline.cpp

```diff
@@ -405,11 +406,13 @@ cmdline_parm almission_arg("-almission", "Autoload multiplayer mission", AT_STRI
 cmdline_parm ingamejoin_arg("-ingame_join", NULL, AT_NONE);	// Cmdline_ingamejoin
 cmdline_parm mpnoreturn_arg("-mpnoreturn", NULL, AT_NONE);	// Cmdline_mpnoreturn  -- Removes 'Return to Flight Deck' in respawn dialog -C
 cmdline_parm objupd_arg("-cap_object_update", "Multiplayer object update cap (0-3)", AT_INT);
+cmdline_parm gateway_ip_arg("-gateway_ip", "Set gateway IP address", AT_STRING);
 
 char *Cmdline_almission = NULL;	//DTP for autoload multi mission.
 int Cmdline_ingamejoin = 0;
 int Cmdline_mpnoreturn = 0;
 int Cmdline_objupd = 3;		// client object updates on LAN by default
+char *Cmdline_gateway_ip = nullptr;
```

Comment:
Please use `SCP_string` for new options since that is easier to use with C++.

Reply:
This requires including pstypes.h in cmdline.h. Is that wanted?

Reply:
Just realized that would duplicate memory too. Might need to rework cmdline code before making the move to `SCP_string`

Reply:
I wouldn't insist on `SCP_string` here.

Reply:
Going to assume this is no longer an issue. In context it's a pointer to an allocated string in memory and not itself a string.